ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Types.hs

```diff
@@ -207,7 +208,14 @@ data ChainDbEnv m blk = CDB
   , cdbBgThreads      :: !(StrictTVar m [Thread m ()])
     -- ^ The background threads.
   , cdbEpochInfo      :: !(EpochInfo m)
-  } deriving (Generic, NoUnexpectedThunks)
+  } deriving (Generic)
+
+-- | We include @blk@ in 'showTypeOf' because it helps resolving type families
+-- (but avoid including @m@ because we cannot impose @Typeable m@ as a
+-- constraint and still have it work with the simulator)
+instance (IOLike m, ProtocolLedgerView blk)
+      => NoUnexpectedThunks (ChainDbEnv m blk) where
+    showTypeOf _ = "ChainDbEnv m " ++ show (typeRep (Proxy @blk))
```

Comment:
Yeah, I meant to ask you about when to override `showTypeOf`, I ignored it so far.